Added Saved Videos tab. For #61.

atomjack opened the pull request.

ram-on commented:
Please document your code by inserting comments.  I'll check out this pull request in the meantime.

atomjack commented:
Ok, I added some comments. I also updated a few other things, mostly to make sure scroll position on the Saved Tab is not lost when you switch back and forth (including when you add or remove a Saved video from one of the other tabs).

ram-on commented:
Just FYI:
- Changed the term from 'Saved' to 'Bookmarkes'.  This is to avoid ambiguity as videos are not saved (technically speaking).
 - The user is now notified whenever he bookmarkes / unbookmarkes a video (via a toast).
